.github/workflows/pkg-deps.yaml

```diff
@@ -0,0 +1,102 @@
+name: Package dependencies
+
+on:
+  workflow_call:
+
+jobs:
+  check-dependency:
+    name: Check dependency
+    runs-on: ubuntu-latest
+    if: |
+      github.event_name == 'pull_request' &&
+      startswith(github.base_ref, 'ubuntu-')
+    env:
+      branch: ${{ github.base_ref }}
+      msg_file: dependencies.message
+    permissions:
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Check changed paths
+        id: changed-paths
+        uses: dorny/paths-filter@v3
+        with:
+          # ref: https://github.com/marketplace/actions/paths-changes-filter
+          filters: |
+            slices:
+              - added|modified: 'slices/**/*.yaml'
+          # Space delimited list usable as command-line argument list in
+          # Linux shell. If needed, it uses single or double quotes to
+          # wrap filename with unsafe characters.
+          list-files: shell
+
+      - name: Check dependencies
+        id: check-deps
+        if: steps.changed-paths.outputs.slices == 'true'
+        run: |
+          set -e
+
+          export LC_COLLATE=C
+
+          version=$(echo "$branch" | grep -Eo '[0-9.]+')
+          docker run -i -d --rm --name ubuntu ubuntu:"$version"
+          docker exec ubuntu apt-get update
+
+          cleanup() {
+            docker stop ubuntu
+          }
+          trap cleanup EXIT
+
+          echo "Slice package dependency issues:" > "$msg_file"
+          for f in ${{ steps.changed-paths.outputs.slices_files }}; do
+            echo "Processing $f.." >&2
+            pkg=$(yq '.package' "$f")
+
+            mapfile -t deps < <(
+              docker exec ubuntu apt depends \
+                --no-recommends --no-suggests --no-conflicts \
+                --no-breaks --no-replaces --no-enhances \
+                "$pkg" 2>/dev/null | \
+              sed -nr 's/.*Depends:\s(\S*).*/\1/p' | \
+              sed 's/<//; s/>//; s/:any//' | \
+              sort | uniq
+            )
+
+            mapfile -t listed < <(
+              yq '.slices.[].essential[]' "$f" | \
+              sed "s/_.*//; /^$pkg$/d" | sort | uniq
+            )
+
+            errs=()
+            for p in "${listed[@]}"; do
+              if printf '%s\0' "${deps[@]}" | grep -Fxqz -- "$p"; then
+                continue
+              fi
+              errs+=("$p")
+            done
+
+            if (( "${#errs[@]}")); then
+              echo "<details>" >> "$msg_file"
+              echo -e "<summary>$f</summary>\n" >> "$msg_file"
+              for e in "${errs[@]}"; do
+                echo "  - \`$e\` is listed in \`essential\`, but is not a dependency." >> "$msg_file"
```

Comment:
would it be possible to have this as a real `diff` instead? This way we don't write any text that might convey whether things are erratic or not

Reply:
How do you mean? `diff` from what?

Reply:
something like

<details><summary>Diff of dependencies</summary>
<p>
```diff
diff --git a/pkg-deps b/slice-deps
index 86e041d..b7b05f8 100644
--- a/pkg-deps
+++ b/slice-deps
@@ -1,3 +1,2 @@
-foo
 bar
-baz
+something-else
```
</p>
</details>

Reply:
I still do not understand to be honest. Do you want something like the following

> Diff of dependencies:
> <details>
> <summary>slices/coreutils.yaml</summary>
> 
> ```diff
> + libcrypt1
> + libgcc-s1
> + libpcre2-8-0
> ```
> </details>

instead of below?

> Diff of dependencies:
> <details>
> <summary>slices/coreutils.yaml</summary>
> 
>   - `libcrypt1` is listed in `essential`, but is not a dependency.
>   - `libgcc-s1` is listed in `essential`, but is not a dependency.
>   - `libpcre2-8-0` is listed in `essential`, but is not a dependency.
>
> </details>

Reply:
Added https://github.com/canonical/chisel-releases/pull/187/commits/a011766c35308ef500f0fad78316e6a6af8ab217. The comment will now look like the following:

> Diff of dependencies:
> 
> <details>
> <summary>slices/coreutils.yaml</summary>
> 
> ```diff
> @@ -1,5 +1,8 @@
>  libacl1
> -libattr1
>  libc6
> +libcrypt1
> +libgcc-s1
>  libgmp10
> +libpcre2-8-0
>  libselinux1
> ```
> 
> </details>
>